### PR TITLE
Use struct on stack instead of opaque struct in opentime

### LINKED ISSUE #3: Manually export individual functions
Manually export individual functions, rather than using nested export "C" blocks. The export macro will be useful later when we add visibility attributes. c.f. https://github.com/PixarAnimationStudios/OpenTimelineIO/commit/5b86f248fd6c661afbb37f400b1248139e7d4984

### LINKED ISSUE #7: Replace TimeInterval* with a pass-by-value TimeInterval struct containing an opaque data packet and associated accessors
Replace TimeInterval* with a pass-by-value TimeInterval struct containing an opaque data packet and associated accessors

### PR DESCRIPTION
**Pull request by KarthikRIyer**
Closes #3 
Closes #7

### CODE REVIEW

**Comment by KarthikRIyer** [comment]
@meshula this is ready for review

File: tests/OTIOClipTests.c [hunk 1]
```diff
@@ -78,96 +72,71 @@ static void otio_clip_constructor_test(void **state) {
...
+    assert_true(OptionalTimeRange_valid(clip_source_range));
+    RationalTime clip_source_range_duration =
+            TimeRange_duration(OptionalTimeRange_value(clip_source_range));
 
     assert_true(RationalTime_equal(clip_source_range_duration, clip_duration));
 
-    assert_true(TimeRange_equal(clip_trimmed_range, clip_source_range));
+    assert_true(TimeRange_equal(clip_trimmed_range, OptionalTimeRange_value(clip_source_range)));
 
-    TimeRange_destroy(tr);
```
Comment (meshula) [inline tests/OTIOClipTests.c:144]: And this chunk shows the payoff of all the work :)

**Review by meshula — COMMENTED** [review]
Fantastic! Some minor edits (which you can click "accept suggestion" to incorporate into the PR). Epic bit of work, thanks

File: tests/OTIOClipTests.c (hunk 1 above)
Comment (KarthikRIyer) [reply tests/OTIOClipTests.c:144]: :relieved:

**Comment by KarthikRIyer** [comment]
Done!

**Review by meshula — APPROVED** [review]
yay!

**Pull request merged**